### ISSUE TITLE
Single-step data connections impossible with auto id issuing

### ISSUE THREAD

**Issue by eettaa** [aws-amplify/amplify-cli#9403]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.18.1

### Amplify CLI Version

7.6.5

### What operating system are you using?

Ubuntu 18.04.5 LTS

### Amplify Categories

api

### Amplify Commands

codegen

### Describe the bug

As far as I can tell, amplify's resolvers do not support creating a parent and child object simultaneously using @hasOne and @belongsTo if we are relying on auto-id generation.

 Schema:
```
type Member @model @auth(rules: [{allow: owner}]) {
  name: String
  settingsName: Settings @hasOne
}

type Settings @model @auth(rules: [{allow: owner}]) {
  darkMode: Boolean
  memberName: Member @belongsTo
}
```
The bulit graphql looks like
```
input CreateMemberInput {
  name: String
  id: ID
  memberSettingsNameId: ID
}
```
and 
```
input CreateSettingsInput {
  darkMode: Boolean
  id: ID
  settingsMemberNameId: ID
}
```
Unless I predetermine the id's of Member and Settings, which I don't if I'm using Amplify's autogen'ed id's, I can't both create these objects and link them in the same step.  This also means that it is impossible to create objects with non-nullable linkage (if one wanted to), because there necessarily would have to be a placeholder value (which is effectively the null value).

### Expected behavior

Similar to how queries are codegen'ed to allow nesting, it would be awesome to be able to have AppSync resolvers support creation commands like
```
CreateMember(name:'Joey', settingsName: {dark_mode: true}) {
  id
  name
  memberSettingsNameId
  settingsName {
    id
    dark_mode
    settingsMemberNameId
  }
}
```
which would result in a returned data object like:
```
{
  id:  'AUTO_ID_A'
  name: 'Joey'
  memberSettingsNameId: 'AUTO_ID_B'
  settingsName:  {
    id: 'AUTO_ID_IB'
    dark_mode: true
    settingsMemberNameId: 'AUTO_ID_A
  }
} 
```
Is there a way to achieve this at present?
I need to do a little more digging to determine whether there is a fundamental resolver limitation here- perhaps there is but it could be solved with a `max_depth` parameter similar to query codegen?

Assuming there is no way to do this at present, are there any recommendations for the best way to safely create/maintain such linkages? I'm worried about e.g. the followup call to create the linkage failing after the first write succeeds, leading to dangling objects, etc.

Thanks for your time and help!!

### Reproduction steps

1) Copy above schema, deploy to cloud
2) Open GraphIQL in App Sync, explore possible creation flows
3) Open local codegen'ed resolver and client libraries, notice that this sort of nesting is not supported.

### GraphQL schema(s)

(relevant schemas included above)

### Log output

logs are irrelevant here

**Comment by edwardfoyle** [comment]
@eettaa you are correct that when using autogenerated IDs it is not possible to create and link two entities in the same mutation operation. To do that you would have to generate the IDs in the frontend code and create the entities with the ID of the linked entity. Using a UUID library should make this relatively straightforward.

**Comment by eettaa** [comment]
Thank you. Is the AppSync (amplify?) auto UUID generator compatible with major UUID issuing libraries?  i.e. if I partially create my own UUID's and partially rely on Amplify, will that be a problem?

**Comment by edwardfoyle** [comment]
@eettaa No that shouldn't be a problem. Any UUID library with sufficient entropy will work
